src/libs/Middleware/SaveResponseInOnyx.ts

```diff
@@ -14,7 +14,7 @@ const SaveResponseInOnyx: Middleware = (requestResponse, request) =>
 
         // Sometimes we call requests that are successfull but they don't have any response or any success/failure data to set. Let's return early since
         // we don't need to store anything here.
-        if (!onyxUpdates && !request.successData && !request.failureData) {
+        if (!onyxUpdates && ((!request.successData && !request.failureData) || !request.finallyData)) {
```

Comment:
Couldn't this just be `&& !request.finallyData` ?

Reply:
I don't believe so, since there would still be cases where `finallyData` is not included (i.e. when `successData` and `failureData` aren't the same) and we'd still want to resolve the promise in that case.

Reply:
Sorry, realizing that my last comment was really ambiguous. what I mean't is, could we do this?

```js
if (!onyxUpdates && !request.successData && !request.failureData && !request.finallyData) {
    return Promise.resolve(response);
}
```

Technically this is not logically equivalent to what you have here, but I think it's what we want.

Imagine a scenario where we have no `onyxUpdates` in the response, we _do_ have `successData` and `failureData`, but we _don't_ have `finallyData`. The expected outcome would be that we _don't_ return early and apply the `successData` or `failureData`. But with what we have here, we would end up returning early:

```
-> !onyxUpdates && ((!request.successData && !request.failureData) || !request.finallyData)
-> !false && ((!true && !true) || !false)
-> true && ((false && false) || true)
-> true && (false || true)
-> true && true
-> true
-> early return ❌ 
```

But if instead we have:

```
-> !onyxUpdates && !request.successData && !request.failureData && !request.finallyData
-> !false && !true && !true && !false
-> true && false && false && true
-> false
-> no early return ✅ 
```

Reply:
Ah, I get what you're saying. Yep, that makes sense.